### PR TITLE
implement a more accurate algorithm to find min_mc_wl in updateStableMcFlag()

### PR DESCRIPTION
**Pull request by guantau**

### CODE REVIEW

File: main_chain.js [hunk 1]
```diff
@@ -420,6 +420,38 @@ function updateMainChain(conn, from_unit, last_added_unit, onDone){
 		});
 	}
 
+	function findMinMcWitnessedLevel(tip_unit, min_level, arrWitnesses, handleMinMcWl){
+		var arrCollectedWitnesses = [];
+		var min_mc_wl = Math.pow(2, 31) - 1;
```
Comment (tonyofbyteball) [inline main_chain.js]: Infinity would be more future-proof

File: main_chain.js [hunk 2]
```diff
@@ -420,6 +420,38 @@ function updateMainChain(conn, from_unit, last_added_unit, onDone){
...
+
+		function addWitnessesAndGoUp(start_unit){
+			storage.readStaticUnitProps(conn, start_unit, function(props){
+				var best_parent_unit = props.best_parent_unit;
+				var level = props.level;
+				if (level === null)
+					throw Error("null level in findMinMcWitnessedLevel");
+				if (level < min_level) {
+					console.log("unit " + start_unit + ", level=" + level + ", first_unstable_levle=" + min_level + ", min_mc_wl=" + min_mc_wl);
```
Comment (tonyofbyteball) [inline main_chain.js]: typo first_unstable_levle

File: main_chain.js [hunk 3]
```diff
@@ -420,6 +420,38 @@ function updateMainChain(conn, from_unit, last_added_unit, onDone){
 		});
 	}
 
+	function findMinMcWitnessedLevel(tip_unit, min_level, arrWitnesses, handleMinMcWl){
```
Comment (tonyofbyteball) [inline main_chain.js]: better use the same name first_unstable_mc_level instead on min_level

File: main_chain.js (hunk 1 above)
Comment (guantau) [reply main_chain.js]: I'll change it to Number.POSTIVE_INFINITY

File: main_chain.js (hunk 2 above)
Comment (guantau) [reply main_chain.js]: ok

File: main_chain.js (hunk 3 above)
Comment (guantau) [reply main_chain.js]: ok

**Pull request closed without merging**